### ISSUE TITLE
[dartdevc] Save location of kernel summary file in ddc metadata

### ISSUE THREAD

**Issue by annagrin** [dart-lang/sdk#44742]
Build systems might produce summary files in a different step from the ddc metadata, so ddc cannot save the location in the metadata for the debugger to consume. Currently the debugger needs to "guess" the names and locations of summary kernel files.

Suggestion: pass a flag to ddc so it can store this information in the metadata, then pass it from the debugger to the expression compiler.

Code from `chrome_proxy_service.dart` with the hack:

```
 for (var module in modules.keys) {
      var serverPath =
          await globalLoadStrategy.serverPathForModule(entrypoint, module);
      dependencies[module] = ModuleInfo(
          p.setExtension(serverPath, '.full.dill'),
          // Replace .ddk to find the summary in google3.
          // TODO: Save location of kernel summary file in ddc metadata.
          p.setExtension(serverPath.replaceAll('.ddk', '.ddc'), '.dill'));
    }
```

**Comment by sigmundch** [comment]
One minor detail to clarify: would be good if the location is encoded in terms of a relative path or something agnostic of the machine location of the build results. 

That's because build systems like bazel require files to be deterministic and stable over time and not include machine-specific details. If we pick a build root folder and use paths relatives to that root, then we would be in a good place.

**Comment by annagrin** [comment]
@sigmundch definitely, I already have an issue for that for other paths. In this case the path will come from the build system so it is already truncated I believe, but I will verify it when I add those to the metadata.

**Comment by sigmundch** [comment]
@annagrin, @nshahan - just curious - is this still planned for this upcoming release?

**Comment by nshahan** [comment]
My recollection from our discussion was that we would prefer not to add the path to the DDC metadata file because DDC does not produce the summary file, and it should be known by the build system before invoking DDC.

@annagrin Does it look like we it would add too much complexity to get the summary file path without involving the metadata?

**Comment by annagrin** [comment]
@nshahan It would be definitely easier if the metadata could contain the summary file path, we currently don't have any other feasible path (ie one that does not require heavy modification of build rules) to get this information without guessing.
